src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,189 @@
+package duke.command;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import duke.DukeException;
+import duke.Storage;
+import duke.Ui;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+/**
+ * Represents an add command. An AddCommand object represents a command
+ * to insert a task into a TaskList. This task can be either a todo, deadline or event.
+ */
+public class AddCommand extends Command {
+
+    /** String array storing the user input. */
+    private String[] userStrings;
```

Comment:
Perhaps this variable can be renamed to `userInputSplit` as `userStrings` sounds a bit unclear and vague.